TodoLogic/src/components/Todo.tsx

```diff
@@ -1,75 +1,128 @@
+  //submitHandler
+  const submitHandler = (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     let date = new Date();
     let currdate = date.toLocaleDateString();
     localStorage.setItem("currdate", currdate);
-  }, []);
+    setShowLabel(true);
+    JSON.parse(localStorage.getItem("submit"));
 
-  const clickHandler = () => {
-    setShowLabel(!showLabel);
-    setBtnVisible(!btnVisible);
+    const submitdata = [
+      {
+        id: Math.random() * 10000,
+        data: input,
+        check: false,
+      },
+    ];
+    setSubmit([...submit, ...submitdata]);
+    setInput("");
   };
 
-  const handleKeyUp = useCallback(
-    (event: KeyboardEvent) => {
-      if (event.key === "Enter") {
-        const element = event.target as HTMLInputElement;
-        setData(element.value);
-      }
-      if (event.key === "Escape") {
-        setBtnVisible(btnVisible);
-        setShowLabel(showLabel);
-      }
-    },
-    [setData, setBtnVisible, setShowLabel]
-  );
+  //changeHandler
+  const changeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const elementValue = e.target as HTMLInputElement;
+    setInput(elementValue.value);
+  };
+  const clickHandler = () => {
+    setShowLabel(true);
+  };
 
   useEffect(() => {
-    if (setData) {
-      window.addEventListener("keyup", handleKeyUp);
+    const d = new Date();
+    const todoData = localStorage.getItem("submit");
+    const newdate = d.toLocaleDateString();
+    const setdate = localStorage.getItem("currdate");
+
+    if (newdate !== setdate) {
+      localStorage.removeItem("submit");
+    }
+    if (todoData) {
+      return JSON.parse(todoData);
     } else {
-      window.removeEventListener("keyup", handleKeyUp);
+      return [];
     }
-    return () => window.removeEventListener("keyup", handleKeyUp);
-  }, [setData, handleKeyUp, submitHandler]);
+  }, [setSubmit]);
+
+  //Esc Key Logic
+  useEffect(() => {
+    const handleEvent = (e: KeyboardEvent) => {
+      if (e.key === "Escape") {
+        setShowLabel(false);
+      }
+    };
+    document.addEventListener("keydown", handleEvent);
+    return () => {
+      document.removeEventListener("keydown", handleEvent);
+    };
+  }, [setShowLabel]);
+
+  useEffect(() => {
+    localStorage.setItem("submit", JSON.stringify(submit));
+  }, [submit]);
 
   return (
     <>
-      <div className="container">
-        <TodoDate />
-        <div>
-          <TodoList value={data} />
-          {showLabel && (
-            <form onSubmit={submitHandler}>
-              <input
-                className="Input-label"
-                type="text"
-                placeholder="Eneter something here"
-                required
-                min="1"
-              />
-            </form>
-          )}
-        </div>
+      <TodoContext.Provider value={{ submit, setSubmit }}>
+        <div className="container">
+          <TodoDate />
+          <div>
+            <TodoList {...submit[0]} />
```

Comment:
Why first item of todolist is passed as props to `TodoList` component. These props are passed but not used inside TodoList component